src/Assignment.ts

```diff
@@ -0,0 +1,59 @@
+import Question from "./model/Question";
+import alasql from "alasql";
+
+const question = new Question(
+    // Prompt for the user
+    `Select all columns from the "Employees" table.`,
+
+    // SQL queries to configure the context
+    `CREATE TABLE Employees (ID INT, Name VARCHAR(255), Supervisor INT);
+    INSERT INTO Employees VALUES
+        (1, "Joe", NULL),
+        (2, "Bill", 1),
+        (3, "Jill", NULL),
+        (4, "Bob", 3),
+        (5, "Marvin", 1);`,
+
+    // Answer key query
+    `SELECT * FROM Employees`);
+
+function compareResults(a: Array<any>, b: Array<any>) {
+    if (a.length != b.length) return false;
+
+    // TODO: This isn't ideal, but it gets the job done for the time being. Write something better in the future.
+    return JSON.stringify(a) === JSON.stringify(b);
+}
+
+export function getQuestion(req: Record<string, any>, res: Record<string, any>) {
+    res.json({prompt: question.prompt, context: question.context});
+}
+
+export function checkAnswer(req: Record<string, any>, res: Record<string, any>) {
+    let result;
+    let userQuery = req.body.query;
+    
+    try {
+        // Create a temporary database using the context queries provided with the question
+        alasql("CREATE DATABASE tempdb; USE tempdb");
```

Comment:
When we use 3rd party services its usually good to have a wrapper for that service rather than calling it directly in our application code. It may be the case that alsql changes parts of their api or that in the future we want to replace it with another service. In these cases, we don't want the 3rd party tool function being called directly in our app because if that change does happen it will lead to large refactors.

So we should probably have some service class that wraps this functionality. That class can then also hide implementation details like exactly the SQL code used to create the DB, we're mixing levels of abstraction by getting bogged down in SQL code for what is a pretty high level function called directly from the API controller.

Reply:
Moved it into a separate file in [this commit](https://github.com/kellerflint/sql-project-api/pull/1/commits/7bfc767e8895d1db472cf00c29fa6322cca6eda7). I also felt like this class warranted more thorough documentation than everything else we've done so far, so I added documentation comments to each method.

Reply:
Yep, looks good!